zap/src/main/java/org/zaproxy/zap/extension/alert/DialogAddAlertTag.java

```diff
@@ -0,0 +1,213 @@
+import org.zaproxy.zap.utils.ZapTextField;
+import org.zaproxy.zap.view.AbstractFormDialog;
+import org.zaproxy.zap.view.LayoutHelper;
+
+public class DialogAddAlertTag extends AbstractFormDialog {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String DIALOG_TITLE =
+            Constant.messages.getString("alert.tags.dialog.add.title");
+    private static final String CONFIRM_BUTTON_LABEL =
+            Constant.messages.getString("alert.tags.dialog.add.button.confirm");
+    private static final String KEY_FIELD_LABEL =
+            Constant.messages.getString("alert.tags.dialog.add.key");
+    private static final String VALUE_FIELD_LABEL =
+            Constant.messages.getString("alert.tags.dialog.add.value");
+    private static final String REPEATED_TAG_KEY_TITLE =
+            Constant.messages.getString("alert.tags.dialog.warning.title.repeated.key");
+    private static final String REPEATED_TAG_KEY_BODY =
+            Constant.messages.getString("alert.tags.dialog.warning.body.repeated.key");
+
+    private ZapTextField keyTextField;
+    private ZapTextArea valueTextArea;
+    private ConfirmButtonValidatorDocListener confirmButtonValidatorDocListener;
+
+    protected AlertTagsTableModel model;
+
+    public DialogAddAlertTag(Dialog owner, AlertTagsTableModel model) {
+        this(owner, model, DIALOG_TITLE);
+    }
+
+    protected DialogAddAlertTag(Dialog owner, AlertTagsTableModel model, String title) {
+        super(owner, title);
+        this.model = model;
+    }
+
+    @Override
+    protected JPanel getFieldsPanel() {
+        JPanel fieldsPanel = new JPanel(new GridBagLayout());
+        JLabel keyLabel = new JLabel(KEY_FIELD_LABEL);
+        JLabel valueLabel = new JLabel(VALUE_FIELD_LABEL);
+
+        int gbcRow = 0;
+        fieldsPanel.add(
+                keyLabel,
+                LayoutHelper.getGBC(
+                        0,
+                        gbcRow,
+                        1,
+                        0,
+                        0.1,
+                        GridBagConstraints.HORIZONTAL,
+                        new Insets(2, 2, 2, 2)));
+        fieldsPanel.add(
+                getKeyTextField(),
+                LayoutHelper.getGBC(
+                        1,
+                        gbcRow,
+                        1,
+                        1,
+                        0.1,
+                        GridBagConstraints.HORIZONTAL,
+                        new Insets(2, 2, 2, 2)));
+        gbcRow++;
+        fieldsPanel.add(
+                valueLabel,
+                LayoutHelper.getGBC(
+                        0,
+                        gbcRow,
+                        1,
+                        0,
+                        0.1,
+                        GridBagConstraints.HORIZONTAL,
+                        new Insets(2, 2, 2, 2)));
+        gbcRow++;
+        fieldsPanel.add(
+                getValueTextArea(),
```

Comment:
Needs a scroll pane, with many chars it starts to expand and move the other field around.

Reply:
Guess this applies to the key as well (when approaching the limit) but just tested with the value.

Reply:
Nope, it's using a text field for the key.